clearwater-queue-manager.root/usr/share/clearwater/clearwater-queue-manager/scripts/get_apply_config_key

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (C) Metaswitch Networks 2016
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+if [ -d /usr/share/clearwater/node_type.d ]; then
+  . /usr/share/clearwater/node_type.d/$(ls /usr/share/clearwater/node_type.d | head -n 1)
+fi
+. /etc/clearwater/config
+
+if [ -v etcd_cluster_key ]; then
```

Comment:
This isn't right. You need to have the `$` in front of etcd_cluster_key. I'm not sure what the `-v` does, but a quick test suggests it's not doing the right thing. Also, you can simplify this check (did you see the comment I left before?) - you can have a check like:

    if [ -z "$etcd_cluster_key" ] || [ $use_single_restart_queue == "Y" ]
      echo -n "apply_config"
    else
      echo -n "apply_config_$etcd_cluster_key"
    fi

If you use this, make sure you check what happens if etcd_cluster_key isn't set at all.